geonode/base/api/serializers.py

```diff
@@ -49,6 +53,21 @@
 logger = logging.getLogger(__name__)
 
 
+class BaseDynamicModelSerializer(DynamicModelSerializer):
+
+    def to_representation(self, instance):
+        data = super(BaseDynamicModelSerializer, self).to_representation(instance)
+        try:
+            path = reverse(self.Meta.view_name)
+            if not path.endswith('/'):
+                path = f"{path}/"
+                url = urljoin(path, str(instance.pk))
```

Comment:
I guess this is overintented

Reply:
@afabiani the problem is that urljoin will consider the first slash it finds in the first path. So it path misses the final slash the resulting url might be truncated.

Reply:
@giohappy there's a syntax error. Like it is now the url and data[link] will be populated only if `not path.endswith('/')`

Reply:
updated

Reply:
oh right @afabiani I thought you were referring to the extra checl.